### PR TITLE
[project-base] docker for production

### PR DESCRIPTION
**Pull request by boris-brtan**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Find the way how to build and deploy project-base for production
|New feature| No 
|BC breaks| No 
|Fixes issues| -
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: project-base/docker/php-fpm/Dockerfile-production [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+ENV LC_ALL=en_US.utf8 LANG=en_US.utf8 LANGUAGE=en_US.utf8
+
+# overwrite the original entry-point from the PHP Docker image with our own
+COPY docker/php-fpm/docker-php-entrypoint /usr/local/bin/
+
+# allow overwriting UID and GID o the user "www-data" to help solve issues with permissions in mounted volumes
+# if the GID is already in use, we will assign GID 82 instead (82 is the standard uid/gid for "www-data" in Alpine)
+ARG www_data_uid
+ARG www_data_gid
+RUN deluser www-data && (addgroup -g $www_data_gid www-data || addgroup -g 82 www-data) && adduser -u $www_data_uid -D -S -G www-data www-data
```
Comment (PetrHeinz) [inline project-base/docker/php-fpm/Dockerfile-production]: This doesn't seem necessary

File: docs/index.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # Shopsys Framework Knowledge Base
 
 ## Introduction
-* [Installation Using Docker](installation/installation-using-docker.md) and [Native Installation](installation/native-installation.md)
+* [Installation Using Docker](installation/installation-using-docker.md) and [Native Installation](installation/native-installation.md) and [Installation Using Docker on Production Server](installation/installation-using-docker-on-production-server.md)
```
Comment (simara-svatopluk) [inline docs/index.md]: The line is quite long. I would prefer a new list item for `Installation Using Docker on Production Server`

File: docs/installation/installation-using-docker-on-production-server.md [hunk 3]
```diff
@@ -1,5 +1,8 @@
 # Production Deployment
```
Comment (simara-svatopluk) [inline docs/installation/installation-using-docker-on-production-server.md]: Please change the title to "Installation .... server"

File: docs/installation/installation-using-docker-on-production-server.md [hunk 4]
```diff
@@ -1,5 +1,8 @@
...
 
+This guide shows us how to install and configure production server applications needed for running [Shopsys/project-base](../../project-base/README.md).
```
Comment (simara-svatopluk) [inline docs/installation/installation-using-docker-on-production-server.md]: * This guide shows <strike>us</strike> how to...
* shopsys/project-base lowercase
* maybe change the link to https://github.com/shopsys/project-base , this readme somewhere in monorepo structure was unexpected for me

File: docs/installation/installation-using-docker-on-production-server.md [hunk 5]
```diff
@@ -1,5 +1,8 @@
...
+We use docker-containers for `php` applications and natively installed applications like `postgres`, `elasticsearch`, `redis` for persisting data without loosing them after new deploys of `project-base`.
```
Comment (simara-svatopluk) [inline docs/installation/installation-using-docker-on-production-server.md]: * maybe docker containers (without dash)
* ... after `project-base` deployments.

File: docs/installation/installation-using-docker-on-production-server.md [hunk 6]
```diff
@@ -1,7 +1,10 @@
 # Installation Using Docker on Production Server
 
-This guide shows how to install and configure production server applications needed for running [shopsys/project-base](https://github.com/shopsys/project-base).
-We use docker containers for `php` applications and natively installed applications like `postgres`, `elasticsearch`, `redis` for persisting data without loosing them after `project-base` deployments.
+This guide shows you how to install and configure production server applications needed for your project based on [Shopsys Framework](https://github.com/shopsys/project-base).  
+We use natively installed applications like postgres, elasticsearch, redis for persisting data without loosing them after deploy of new version of the project.
+We do not want to setup each application or each micro-service application manually and in server environment and have some separation between them.
```
Comment (vitek-rostislav) [inline docs/installation/installation-using-docker-on-production-server.md]: I would use this sentence right after the first one ("This guide shows you how to install and configure production server applications needed for your project based on Shopsys Framework").

Comment (vitek-rostislav) [reply docs/installation/installation-using-docker-on-production-server.md]: also, ~~micro-service~~ -> microservice

Comment (vitek-rostislav) [reply docs/installation/installation-using-docker-on-production-server.md]: also, the part "...and in server environment and have some separation between them." is unclear to me.

File: docs/installation/installation-using-docker-on-production-server.md [hunk 7]
```diff
@@ -1,7 +1,10 @@
 # Installation Using Docker on Production Server
 
-This guide shows how to install and configure production server applications needed for running [shopsys/project-base](https://github.com/shopsys/project-base).
-We use docker containers for `php` applications and natively installed applications like `postgres`, `elasticsearch`, `redis` for persisting data without loosing them after `project-base` deployments.
+This guide shows you how to install and configure production server applications needed for your project based on [Shopsys Framework](https://github.com/shopsys/project-base).  
+We use natively installed applications like postgres, elasticsearch, redis for persisting data without loosing them after deploy of new version of the project.
```
Comment (vitek-rostislav) [inline docs/installation/installation-using-docker-on-production-server.md]: This sentence should be before the last one ("This guide also shows you how to setup first...")

Comment (vitek-rostislav) [reply docs/installation/installation-using-docker-on-production-server.md]: I would rewrite the sentence in this fashion: "As we do not want to lose data after deploying a new version of the project, we install all the data storages (postgres, elasticsearch, redis) natively though."

**Pull request merged**